From 3cba9710d143982cf01cba934d625b2315cac734 Mon Sep 17 00:00:00 2001
From: huzhen <george.hu@spacemit.com>
Date: Thu, 17 Oct 2024 15:04:09 +0800
Subject: [PATCH 1182/1448] rtl8852be: Initialize the internal buf of the
 function

Solve the problem of incorrect end character of cmd_str

Change-Id: I447ad2fa4dbe476e885cbef2f54ead9b6da7af1b
---
 drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_proc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_proc.c b/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_proc.c
index ede8e678fff6..9b679c8e9b59 100644
--- a/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_proc.c
+++ b/drivers/net/wireless/realtek/rtl8852be/os_dep/linux/rtw_proc.c
@@ -6118,6 +6118,7 @@ static int proc_get_self_diag_info(struct seq_file *m, void *v)
 		RTW_PRINT_SEL(m, "\n==================================[MAC info]==================================\n");
 		cmd_string = "wdt_log 3E";
 		cmd_len = strlen(cmd_string);
+		_rtw_memset(buf, 0, 25);
 		_rtw_memcpy(buf, cmd_string,  cmd_len);
 		proc_set_phydm_cmd(NULL, buf, cmd_len, NULL, dev, RTW_PROC_CMD_MAC, out, out_len);
 		proc_get_phydm_cmd(m, v, out, out_len);
@@ -6163,6 +6164,7 @@ static int proc_get_self_diag_info(struct seq_file *m, void *v)
 
 		cmd_string = "wdt_log 0";
 		cmd_len = strlen(cmd_string);
+		_rtw_memset(buf, 0, 25);
 		_rtw_memcpy(buf, cmd_string,  cmd_len);
 		proc_set_phydm_cmd(NULL, buf, cmd_len, NULL, dev, RTW_PROC_CMD_MAC, phydm_msg, PHYDM_MSG_LEN);
 	}
-- 
2.47.0

